src/compas_timber/connections/ball_node.py

```diff
@@ -0,0 +1,102 @@
+from compas.geometry import Point
+
+from compas_timber.elements import BallNodeFastener
+from compas_timber.utils import intersection_line_line_param
+
+from .joint import Joint
+from .solver import JointTopology
+
+
+class BallNodeJoint(Joint):
+    """Represents a ball node type joint which joins the ends of multiple beams,
```

Comment:
where does this sit in the whole discussion of "fastener is a result of the design" vs. "design must comply with pre-defined fastener"?

Reply:
100% Fasteners respond to design.